### PR TITLE
kaitai-struct-compiler: use shipped launch wrapper

### PR DESCRIPTION
**Pull request by fridtjof**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Review by Adsada2205 — APPROVED** [review]

**Comment by fridtjof** [comment]
Here we go; Bintray was broken yesterday. Should be ready to merge now

**Comment by fridtjof** [comment]
Can someone please look at this? My changes were approved, but never merged - the current recipe breaks a related tool on macOS using Homebrew as can be seen in the issue referenced

**Comment by GreyCat** [comment]
This is still applicable and important change. Someone with commit rights to this repo, please merge in?

**Comment by fridtjof** [comment]
As you can see [here](https://github.com/orgs/kaitai-io/people), @GreyCat is one of the main authors of the kaitai project (which includes both kaitai-struct-compiler and kaitai-struct-visualizer, which is relevant to this PR). He approved this change/fix, so can you please merge this?

**Comment by fridtjof** [comment]
@Adsada2205 you approved this PR on Mar 30, can you merge it, please? I'd really appreciate it :)

File: Formula/kaitai-struct-compiler.rb [hunk 1]
```diff
@@ -3,18 +3,17 @@ class KaitaiStructCompiler < Formula
...
 
   def install
     libexec.install Dir["lib/*"]
+    bin.install "bin/kaitai-struct-compiler"
 
-    (bin/"kaitai-struct-compiler").write <<~EOS
-      #!/bin/bash
-      java -cp "#{libexec}/*" io.kaitai.struct.JavaMain "$@"
-    EOS
+    inreplace bin/"kaitai-struct-compiler", /^declare -r lib_dir=.*$/, "declare -r lib_dir='#{libexec}'"
```
Comment (ilovezfs) [inline Formula/kaitai-struct-compiler.rb]: Why is the inreplace necessary? If we install kaitai-struct-compiler in `libexec/bin`, and then add a symlink in `bin` is the `inreplace` still needed?

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/kaitai-struct-compiler.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/kaitai-struct-compiler.rb]: @fridtjof ping

**Review by fridtjof — COMMENTED** [review]
> Would lib.install Dir["lib/*"] work for this? I could not find documentation on this.

Would this work, like bin.install works?

Edit: oh snap, i hit the wrong reply button

**Comment by ilovezfs** [comment]
Probably `libexec.install Dir["*"]` is what you want, followed by symlinking into `bin`.

**Comment by fridtjof** [comment]
Like this? I tested it, and it works

**Comment by ilovezfs** [comment]
yup. much better!

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks again @fridtjof!

**Pull request merged**